elements/userlevel/todpdkdevice.cc

```diff
@@ -124,7 +124,16 @@ inline struct rte_mbuf* get_mbuf(Packet* p, bool create=true) {
 
     if (likely(DPDKDevice::is_dpdk_packet(p))) {
         mbuf = (struct rte_mbuf *) p->destructor_argument();
-        p->set_buffer_destructor(DPDKDevice::fake_free_pkt);
+        rte_pktmbuf_pkt_len(mbuf) = p->length();
+        rte_pktmbuf_data_len(mbuf) = p->length();
+        if (p->shared()) {
+            /*Prevent DPDK from freeing the buffer. When all shared packet
+             * are freed, DPDKDevice::free_pkt will effectively destroy it.*/
+            rte_mbuf_refcnt_update(mbuf, 1);
```

Comment:
I'm trying to understand how DPDKDevice::free_pkt will be called on the last packet, as the destructor is removed below with p->reset_buffer()? Shouldn't you decrement the DPDK reference counter if it's the last packet and then let DPDK free the buffer instead?